packages/lisk-p2p/src/peer_pool.ts

```diff
@@ -176,7 +177,14 @@ export class PeerPool extends EventEmitter {
 			return false;
 		}
 
-		this.addPeer({ ...peerInfo }, socket);
+		this.addPeer(
+			{
+				ipAddress: peerInfo.ipAddress,
+				wsPort: peerInfo.wsPort,
+				height: peerInfo.height,
+			},
```

Comment:
We should pass as many fields as possible. In the case of an inbound connection, we have enough data from the socket to supply a DiscoveredPeerInfo. Inbound peers should be considered as discovered since all their peer info is passed along with the inbound connection.

Not sure what the solution is; maybe we need a separate method like `peerPool.addDiscoveredPeer`?

Reply:
yeah this is a bit tricky, I guess we should have separate method to add peer in peer_pool because it will get complicated to handle two types in `addPeer()`.
So if we will have a separate `addDiscoveredPeer` then we can explicitly call `updatePeerInfo`to update `detailedPeerInfo` of Peer as its constructor only accepts `P2PPeerInfo`